torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -38,6 +44,30 @@ using namespace torch::autograd;
 
 namespace torch { namespace utils {
 
+bool is_numpy_available() {
+  static bool available = []() {
```

Comment:
Is it also possible to check if we're ABI compatible with numpy here as well? To avoid the 1.20 -> 1.19 issue we're observing?

Reply:
There's no good way of testing actual ABI compat. And NumPy should already refuse to import if it finds a runtime version that is lower than build-time version.

> To avoid the 1.20 -> 1.19 issue we're observing?

I assume this was a packaging bug? Despite some confusion on a TensorFlow issue, there was no ABI break in 1.19 or 1.20 as far as I know.

Reply:
I think first step in that direction is to preserve numpy version PyTorch was linked against in `torch.__config__`, will add in in a separate PR. But based on quick smoke check that I run, 1.19 is forward compatible with 1.20, we just need to specify oldest version we want to support in requirements.